network/transparent_vlan_endpointclient_linux.go

```diff
@@ -310,6 +310,16 @@ func (client *TransparentVlanEndpointClient) PopulateVM(epInfo *EndpointInfo) er
 		logger.Info("Failed to parse the mac address", zap.String("defaultHostVethHwAddr", defaultHostVethHwAddr))
 	}
 
+	// Proactively clean up any leftover veth interfaces before creating new ones
+	if err = client.netlink.DeleteLink(client.vnetVethName); err != nil {
+		logger.Info("Could not proactively clean up vnet veth",
```

Comment:
How does this cleanup different from the one we have during the ADD call ? It can happen at this point to ?referring to this:

> The above shouldn't be necessary since we already validate the existence of the veth interfaces on ADD, but it seems like sometimes the veth creation can pass validation, but then disappear for a short period of time before re-appearing, bypassing the cleanup logic during the add.

Curious to know if we know the reason for disappearing for a short period of time.

Reply:
yes we believe the delete link could silently fail, but if that happens, the next call to create the veth pair will fail the ADD call (file exists... ) and then the container runtime will retry, and eventually the leaked interfaces should be deleted. we are investigating the reason for the disappearance with the azure linux team at the moment.